pkg/roachprod/testselector/roachtestselector/consumer.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+// consumer is responsible for reading the csv files containing the test details generated by testselector
+
+package roachtestselector
+
+import (
+	"bytes"
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+
+	"cloud.google.com/go/storage"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/errors"
+	"google.golang.org/api/option"
+)
+
+// ReadTestsToRun reads the tests to run based on certain criteria:
+// 1. the number of time a test has been successfully running
+// 2. the test is new
+// 3. the test has not been run for a while
+// 4. a subset of the successful tests
+// It returns a new list of selected tests. The individual tests in the input
+// are also modified with the skip reason.
+// It also returns the number of tests that satisfied the selection criteria.
+func ReadTestsToRun(
+	ctx context.Context, tests []registry.TestSpec, cloud, suite string,
+) ([]registry.TestSpec, int, error) {
+	options := []option.ClientOption{option.WithScopes(storage.ScopeReadOnly), option.WithQuotaProject(project)}
+	cj := os.Getenv("GOOGLE_EPHEMERAL_CREDENTIALS")
+	if len(cj) != 0 {
+		options = append(options, option.WithCredentialsJSON([]byte(cj)))
+	} else {
+		fmt.Printf("GOOGLE_EPHEMERAL_CREDENTIALS env is not set.\n")
+	}
+	client, err := storage.NewClient(ctx, options...)
+	if err != nil {
+		return tests, len(tests), errors.NewAssertionErrorWithWrappedErrf(err, "connection to GCS failed")
+	}
+	defer func() { _ = client.Close() }()
+
+	object := fmt.Sprintf("%s-%s-%s.%s", testsFileLocation, suite, cloud, testsCsvExtension)
+	r, err := client.Bucket(bucket).Object(object).NewReader(ctx)
+	if err != nil {
+		return tests, len(tests), errors.NewAssertionErrorWithWrappedErrf(err,
+			"failed to get the object %s in bucket %s", object, bucket)
+	}
+	defer func() { _ = r.Close() }()
+	body, err := io.ReadAll(r)
+	if err != nil {
+		return tests, len(tests), errors.NewAssertionErrorWithWrappedErrf(err, "failed to read CSV from GCS")
+	}
+	cr := csv.NewReader(bytes.NewReader(body))
+	data, err := cr.ReadAll()
+	if err != nil {
+		return tests, len(tests), errors.NewAssertionErrorWithWrappedErrf(err, "failed to read CSV data")
+	}
+	testNamesToRun := make(map[string]string)
```

Comment:
It seems like the value of this map is always supposed to be "yes" or "no". So Optional: make this a `map[string]bool` instead, and just translate the `yes` and `no` to `true` or `false` when you initialize the map.

Reply:
I will change this to have all the parameters from csv captured.